documentation/library_guide/extension_api.rst

```diff
@@ -313,3 +313,68 @@ Example of Range-based API usage
 
         copy(oneapi::dpl::execution::dpcpp_default, view, range_res);
     }
+
+Async API
+--------------------------
+
+Functions defined in the STL ``<algorithm>`` or ``<numeric>`` headers are traditionally blocking. oneDPL extends the functionality of C++17 parallel algorithms by providing asynchronous algorithm with non-blocking behavior. This experimental feature enables expressing concurrent control flow by building dependency chains and interleaving algorithm calls, as well as interoperability with SYCL kernels. 
+
+The current implementation for async algorithms is limited to DPC++ Execution Policies.
+All functionality described below is available in ``oneapi::dpl::experimental`` namespace.
+
+The following async algorithms are currently supported:
+
+* ``copy_async``
+* ``fill_async``
+* ``for_each_async``
+* ``reduce_async``
+* ``transform_async``
+* ``transform_reduce_async``
+* ``sort_async``
+
+All interfaces listed above are a subset of C++17 STL algorithms
+with the suffix ``_async`` added to the corresponding name (e.g. ``reduce``, ``sort``, ... ). The behavior and signatures are overlapping with the C++17 STL algorithm with the following changes:
+
+* Do not block the execution.
+* Take an arbitrary number of events (including 0) as last arguments to allow expressing input dependencies.
+* Return future-like object that allows ``wait`` for completion and ``get`` the result.
+
+The type of the future-like object returned from asynchronous algorithm is unspecified. The following member functions are present:
+
+* ``get()``: returns the result.
+* ``wait()``: wait for the result to become available.
+
+If the returned object is the result of an algorithm with device policy, it can be converted into a ``sycl::event``. Lifetime of any resources the algorithm allocates (e.g. temporary storage) is bound to the lifetime of the returned object.
+
+Utility functions:
+
+* ``wait_for_all(…)``: wait for an arbitrary number of ``sycl::event`` to become ready.
+
+
+Example of Async API usage
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: cpp
+
+    #include <oneapi/dpl/execution>
+    #include <oneapi/dpl/async>
+    #include <CL/sycl.hpp>
+    
+    int main() {
+        using namespace oneapi;
+        {
+            /* Build and compute a simple dependency chain: Fill buffer -> Transform -> Reduce */
+            sycl::buffer<int> a{10};
+            
+            auto my_policy = dpl::execution::dpcpp_default;
```

Comment:
Do we need this line? I understand the idea behind that. You don't want to put `dpl::execution::dpcpp_default` in each algorithm call. I am not strongly suggesting direct passing without additional variable but probably it would look better. If you disagree fill free to resolve this conversation.